### ISSUE TITLE
Path issue related to sonar.ts.tslintpath (Version 9) - work fine in version 8

### ISSUE THREAD

**Issue by edychandra** [Pablissimo/SonarTsPlugin#47]
Hi @Pablissimo. 
Thanks for writing this plugins.
I have this issue where whenever I set the sonar.ts.tslintpath, I get this error:
> 16:40:23 INFO: Sensor com.pablissimo.sonar.TsLintSensor
> 16:40:23 ERROR: TsLint Err: module.js:327
> 16:40:23 ERROR: TsLint Err:     throw err;
> 16:40:23 ERROR: TsLint Err:     ^
> 16:40:23 ERROR: TsLint Err: 
> 16:40:23 ERROR: TsLint Err: Error: Cannot find module '/var/jenkins_home/workspace/Test_Customer_Service/"/var/jenkins_home/workspace/Test_Customer_Service/node_modules/tslint/bin/tslint"'
> 16:40:23 ERROR: TsLint Err:     at Function.Module._resolveFilename (module.js:325:15)
> 16:40:23 ERROR: TsLint Err:     at Function.Module._load (module.js:276:25)
> 16:40:23 ERROR: TsLint Err:     at Function.Module.runMain (module.js:441:10)
> 16:40:23 ERROR: TsLint Err:     at startup (node.js:134:18)
> 16:40:23 ERROR: TsLint Err:     at node.js:962:3

It seems like it adds my project path to the path I set in sonar properties while also adding double quote. 
I also try no to set the sonar.ts.tslintpath but it gives me this warning:

> 15:46:55 INFO: Sensor com.pablissimo.sonar.TsLintSensor
> 15:46:55 WARN: Path to tslint not defined or not found. Skipping tslint analysis.
> 15:46:55 INFO: Sensor com.pablissimo.sonar.TsLintSensor (done) | time=0ms

It then seems to skip the analysis. (Tested this by introducing linting issue, the analysis result still shows no issue) My tslint path is matching the default In TsLintSensor.java, TSLINT_FALLBACK_PATH. I understand that the plugin has not been tested on Linux, could this be Linux specific?

**Comment by Pablissimo** [comment]
It might be, but the quotes are odd. Can you try the following please?

* Tell me what version of the plugin you're using (latest is 0.9)
- If you're on 0.9 try downgrading to 0.8 - there were changes that went in recently to try to make all paths absolute which may be bugging out
* Can you post your sonar-project.properties file (or at least the exact line that configures sonar.ts.tslintpath?
* Are you quoting the path in either sonar-project.properties or in the admin portal? You shouldn't need quotes (it shouldn't blow up encountering them, but try removing them if you have added them)

**Comment by Pablissimo** [comment]
Also - if you try running the sonar-runner with the -X -e switches and see what the debug output is? It'll tell you where it's searching for tslint when you haven't specified a path to it, as the structure you describe above should just have tslint automatically found and run without a problem.

**Comment by edychandra** [comment]
@Pablissimo, Awesome downgraded to 0.8 and it works. This makes me day. :+1: :+1: 
Thank you very much!!!

I reverted back to 0.9 and run sonar runner with -X -e switches, log is below. Also the properties file. (GIthub doesnt allow me to upload it)

**Error log:** 
> 10:53:43 [Test_Customer_Service] $ /tools/hudson.plugins.sonar.SonarRunnerInstallation/ADOP_SonarRunner_2.4/bin/sonar-runner -X -e -Dsonar.jdbc.url=jdbc:<REMOVED> ******** ******** -Dsonar.host.url=<REMOVED> ******** ******** -Dsonar.projectBaseDir=/workspace/Test_Customer_Service -Dsonar.sourceEncoding=UTF-8 -Dsonar.ts.tslintconfigpath=tslint.json -Dsonar.language=ts -Dsonar.sources=. -Dsonar.login=admin -Dsonar.exclusions=node_modules/**,typings/**,test/record/** -Dsonar.projectVersion=1.0.0 -Dsonar.projectKey=Test.Customer.Service "-Dsonar.projectDescription=Project test results." -Dsonar.password=<REMOVED> -Dsonar.ts.excludetypedefinitionfiles=true "-Dsonar.projectName=TEST CUSTOMER SERVICE"
> 10:53:43 WARN: sonar-runner script is deprecated. Please use sonar-scanner instead.
> 10:53:44 INFO: Scanner configuration file: /tools/hudson.plugins.sonar.SonarRunnerInstallation/ADOP_SonarRunner_2.4/conf/sonar-scanner.properties
> 10:53:44 INFO: Project root configuration file: NONE
> 10:53:44 INFO: SonarQube Scanner 2.6.1
> 10:53:44 INFO: Java 1.8.0_45 Oracle Corporation (64-bit)
> 10:53:44 INFO: Linux 3.10.0-229.14.1.el7.x86_64 amd64
> 10:53:44 INFO: Error stacktraces are turned on.
> 10:53:44 INFO: User cache: /root/.sonar/cache
> 10:53:44 DEBUG: Extract sonar-scanner-api-batch in temp...
> 10:53:44 DEBUG: Get bootstrap index...
> 10:53:44 DEBUG: Download: http://<REMOVED>/sonar/batch_bootstrap/index
> 10:53:44 DEBUG: Get bootstrap completed
> 10:53:44 DEBUG: Create isolated classloader...
> 10:53:44 DEBUG: Start temp cleaning...
> 10:53:44 DEBUG: Temp cleaning done
> 10:53:44 DEBUG: Execution getVersion
> 10:53:44 DEBUG: Execution start
> 10:53:45 DEBUG: Publish global mode
> 10:53:45 DEBUG: cache: /root/.sonar/ws_cache/http%3A%2F%2F<REMOVED>%3A9000%2Fsonar/global
> 10:53:45 INFO: Load global repositories
> 10:53:46 DEBUG: GET 200 http://<REMOVED>/sonar/batch/global | time=171ms
> 10:53:46 INFO: Load global repositories (done) | time=240ms
> 10:53:46 WARN: Property 'sonar.jdbc.url' is not supported any more. It will be ignored. There is no longer any DB connection to the SQ database.
> 10:53:46 WARN: Property 'sonar.jdbc.username' is not supported any more. It will be ignored. There is no longer any DB connection to the SQ database.
> 10:53:46 WARN: Property 'sonar.jdbc.password' is not supported any more. It will be ignored. There is no longer any DB connection to the SQ database.
> 10:53:46 INFO: User cache: /root/.sonar/cache
> 10:53:46 INFO: Load plugins index
> 10:53:46 DEBUG: GET 200 http://<REMOVED>/sonar/deploy/plugins/index.txt | time=3ms
> 10:53:46 INFO: Load plugins index (done) | time=4ms
> 10:53:46 DEBUG: Load plugins
> 10:53:46 DEBUG: Load plugins (done) | time=232ms
> 10:53:46 DEBUG: API compatibility mode is enabled on plugin Dependency-Check [dependencycheck] (built with API lower than 5.2)
> 10:53:46 DEBUG: API compatibility mode is enabled on plugin Java [java] (built with API lower than 5.2)
> 10:53:46 DEBUG: API compatibility mode is enabled on plugin Git [scmgit] (built with API lower than 5.2)
> 10:53:46 DEBUG: API compatibility mode is enabled on plugin ZAP [zap] (built with API lower than 5.2)
> 10:53:46 DEBUG: API compatibility mode is enabled on plugin TypeScript [typescript] (built with API lower than 5.2)
> 10:53:46 DEBUG: API compatibility mode is enabled on plugin Build Breaker [buildbreaker] (built with API lower than 5.2)
> 10:53:46 DEBUG: API compatibility mode is enabled on plugin Generic Coverage [genericcoverage] (built with API lower than 5.2)
> 10:53:46 DEBUG: Plugins:
> 10:53:46 DEBUG:   * Dependency-Check 1.0.3 (dependencycheck)
> 10:53:46 DEBUG:   * Java 3.14 (java)
> 10:53:46 DEBUG:   * LDAP 1.5.1 (ldap)
> 10:53:46 DEBUG:   * Git 1.2 (scmgit)
> 10:53:46 DEBUG:   * ZAP 1.0.0-0 (zap)
> 10:53:46 DEBUG:   * TypeScript 0.9-SNAPSHOT (typescript)
> 10:53:46 DEBUG:   * Build Breaker 1.1 (buildbreaker)
> 10:53:46 DEBUG:   * Generic Coverage 1.2 (genericcoverage)
> 10:53:46 DEBUG: Execution getVersion
> 10:53:46 INFO: SonarQube server 5.5
> 10:53:46 INFO: Default locale: "en_US", source code encoding: "UTF-8"
> 10:53:46 DEBUG: Work directory: /workspace/Test_Customer_Service/.sonar
> 10:53:46 DEBUG: Execution getVersion
> 10:53:46 DEBUG: Execution execute
> 10:53:47 INFO: Process project properties
> 10:53:47 DEBUG: Process project properties (done) | time=3ms
> 10:53:47 DEBUG: cache: /root/.sonar/ws_cache/http%3A%2F%2F<REMOVED>%3A9000%2Fsonar/5.5/projects/Test.Customer.Service
> 10:53:47 INFO: Load project repositories
> 10:53:47 DEBUG: GET 200 http://<REMOVED>/sonar/batch/project.protobuf?key=Test.Customer.Service | time=89ms
> 10:53:47 INFO: Load project repositories (done) | time=154ms
> 10:53:47 DEBUG: Available languages:
> 10:53:47 DEBUG:   * Neutral => "neutral"
> 10:53:47 DEBUG:   * Java => "java"
> 10:53:47 DEBUG:   * TypeScript => "ts"
> 10:53:47 INFO: Load quality profiles
> 10:53:47 DEBUG: GET 200 http://<REMOVED>/sonar/api/qualityprofiles/search.protobuf?projectKey=Test.Customer.Service | time=83ms
> 10:53:47 INFO: Load quality profiles (done) | time=88ms
> 10:53:47 INFO: Load active rules
> 10:53:48 DEBUG: GET 200 http://<REMOVED>/sonar/api/rules/search.protobuf?f=repo,name,severity,lang,internalKey,templateKey,params,actives&activation=true&qprofile=java-sonar-way-85046&p=1&ps=500 | time=870ms
> 10:53:48 DEBUG: GET 200 http://<REMOVED>/sonar/api/rules/search.protobuf?f=repo,name,severity,lang,internalKey,templateKey,params,actives&activation=true&qprofile=neutral-neutral-69159&p=1&ps=500 | time=143ms
> 10:53:48 DEBUG: GET 200 http://<REMOVED>/sonar/api/rules/search.protobuf?f=repo,name,severity,lang,internalKey,templateKey,params,actives&activation=true&qprofile=ts-tslint-62563&p=1&ps=500 | time=203ms
> 10:53:48 INFO: Load active rules (done) | time=1272ms
> 10:53:48 INFO: Publish mode
> 10:53:48 DEBUG: Start recursive analysis of project modules
> 10:53:48 INFO: -------------  Scan TEST CUSTOMER SERVICE
> 10:53:49 INFO: Language is forced to ts
> 10:53:49 INFO: Load server rules
> 10:53:49 DEBUG: GET 200 http://<REMOVED>/sonar/api/rules/list.protobuf | time=95ms
> 10:53:49 INFO: Load server rules (done) | time=103ms
> 10:53:49 DEBUG: Initializers : 
> 10:53:49 INFO: Base dir: /workspace/Test_Customer_Service
> 10:53:49 INFO: Working dir: /workspace/Test_Customer_Service/.sonar
> 10:53:49 INFO: Source paths: .
> 10:53:49 INFO: Source encoding: UTF-8, default locale: en_US
> 10:53:49 INFO: Index files
> 10:53:49 INFO: Excluded sources: 
> 10:53:49 INFO:   node_modules/**
> 10:53:49 INFO:   typings/**
> 10:53:49 INFO:   test/record/**
> 10:53:49 DEBUG: Declared extensions of language Neutral were converted to sonar.lang.patterns.neutral : 
> 10:53:49 DEBUG: Declared extensions of language Java were converted to sonar.lang.patterns.java : **/*.java,**/*.jav
> 10:53:49 DEBUG: Declared extensions of language TypeScript were converted to sonar.lang.patterns.ts : **/*.ts,**/*.tsx
> 10:53:49 DEBUG: Language of file 'app.ts' is detected to be 'ts'
> 10:53:49 DEBUG: Language of file 'controllers/feedback.ts' is detected to be 'ts'
> 10:53:49 DEBUG: Language of file 'controllers/plan.ts' is detected to be 'ts'
> 10:53:49 DEBUG: Language of file 'models/feedback.ts' is detected to be 'ts'
> 10:53:49 DEBUG: Language of file 'models/plan.ts' is detected to be 'ts'
> 10:53:49 DEBUG: Language of file 'server.ts' is detected to be 'ts'
> 10:53:49 DEBUG: Language of file 'services/feedback.ts' is detected to be 'ts'
> 10:53:49 DEBUG: Language of file 'services/plan.ts' is detected to be 'ts'
> 10:53:49 DEBUG: Language of file 'test/feedback.ts' is detected to be 'ts'
> 10:53:49 DEBUG: Language of file 'test/plan.ts' is detected to be 'ts'
> 10:53:49 INFO: 10 files indexed
> 10:53:49 INFO: 8741 files ignored because of inclusion/exclusion patterns
> 10:53:49 INFO: Quality profile for ts: tslint
> 10:53:49 INFO: JaCoCoSensor: JaCoCo report not found : /workspace/Test_Customer_Service/target/jacoco.exec
> 10:53:49 INFO: JaCoCoItSensor: JaCoCo IT report not found: /workspace/Test_Customer_Service/target/jacoco-it.exec
> 10:53:49 DEBUG: Code colorizer, supported languages: 
> 10:53:49 DEBUG: Sensors : Lines Sensor (wrapped) -> ZapSensor -> com.pablissimo.sonar.TsLintSensor@2c22a348 -> LOCSensor -> TsCoverageSensor -> SCM Sensor (wrapped) -> Zero Coverage Sensor (wrapped) -> Code Colorizer Sensor (wrapped) -> CPD Block Indexer (wrapped)
> 10:53:49 INFO: Sensor Lines Sensor (wrapped)
> 10:53:49 INFO: Sensor Lines Sensor (wrapped) (done) | time=13ms
> 10:53:49 INFO: Sensor ZapSensor
> 10:53:49 WARN: ZAP report not found at target/zap-reports/zapReport.html
> 10:53:49 INFO: Sensor ZapSensor (done) | time=0ms
> 10:53:49 INFO: Sensor com.pablissimo.sonar.TsLintSensor
> 10:53:49 DEBUG: Path sonar.ts.tslintpath not specified, falling back to node_modules/tslint/bin/tslint
> 10:53:49 DEBUG: Found sonar.ts.tslintconfigpath Lint path to be 'tslint.json'
> 10:53:49 DEBUG: Path sonar.ts.tslintrulesdir not specified, falling back to null
> 10:53:49 DEBUG: Split 10 files into 1 batches for processing
> 10:53:49 DEBUG: Executing TsLint with command: node "/workspace/Test_Customer_Service/node_modules/tslint/bin/tslint" --format json --config "/workspace/Test_Customer_Service/tslint.json" "/workspace/Test_Customer_Service/app.ts" "/workspace/Test_Customer_Service/controllers/feedback.ts" "/workspace/Test_Customer_Service/controllers/plan.ts" "/workspace/Test_Customer_Service/models/feedback.ts" "/workspace/Test_Customer_Service/models/plan.ts" "/workspace/Test_Customer_Service/server.ts" "/workspace/Test_Customer_Service/services/feedback.ts" "/workspace/Test_Customer_Service/services/plan.ts" "/workspace/Test_Customer_Service/test/feedback.ts" "/workspace/Test_Customer_Service/test/plan.ts"
> 10:53:50 ERROR: TsLint Err: module.js:327
> 10:53:50 ERROR: TsLint Err:     throw err;
> 10:53:50 ERROR: TsLint Err:     ^
> 10:53:50 ERROR: TsLint Err: 
> 10:53:50 ERROR: TsLint Err: Error: Cannot find module '/workspace/Test_Customer_Service/"/workspace/Test_Customer_Service/node_modules/tslint/bin/tslint"'
> 10:53:50 ERROR: TsLint Err:     at Function.Module._resolveFilename (module.js:325:15)
> 10:53:50 ERROR: TsLint Err:     at Function.Module._load (module.js:276:25)
> 10:53:50 ERROR: TsLint Err:     at Function.Module.runMain (module.js:441:10)
> 10:53:50 ERROR: TsLint Err:     at startup (node.js:134:18)
> 10:53:50 ERROR: TsLint Err:     at node.js:962:3
> 10:53:50 INFO: Sensor com.pablissimo.sonar.TsLintSensor (done) | time=218ms
> 10:53:50 INFO: Sensor LOCSensor
> 10:53:50 INFO: Sensor LOCSensor (done) | time=18ms
> 10:53:50 INFO: Sensor TsCoverageSensor
> 10:53:50 INFO: Sensor TsCoverageSensor (done) | time=3ms
> 10:53:50 INFO: Sensor SCM Sensor (wrapped)
> 10:53:50 INFO: Sensor SCM Sensor (wrapped) (done) | time=0ms
> 10:53:50 INFO: Sensor Zero Coverage Sensor (wrapped)
> 10:53:50 INFO: Sensor Zero Coverage Sensor (wrapped) (done) | time=10ms
> 10:53:50 INFO: Sensor Code Colorizer Sensor (wrapped)
> 10:53:50 INFO: Sensor Code Colorizer Sensor (wrapped) (done) | time=2ms
> 10:53:50 INFO: Sensor CPD Block Indexer (wrapped)
> 10:53:50 INFO: DefaultCpdBlockIndexer is used for ts
> 10:53:50 DEBUG: No CpdMapping for language ts
> 10:53:50 INFO: Sensor CPD Block Indexer (wrapped) (done) | time=0ms
> 10:53:50 INFO: Calculating CPD for 0 files
> 10:53:50 INFO: CPD calculation finished
> 10:53:50 INFO: Analysis report generated in 63ms, dir size=22 KB
> 10:53:50 INFO: Analysis reports compressed in 13ms, zip size=14 KB
> 10:53:50 DEBUG: Upload report
> 10:53:50 DEBUG: POST 200 http://<REMOVED>/sonar/api/ce/submit?projectKey=Test.Customer.Service&projectName=TEST%20CUSTOMER%20SERVICE | time=73ms
> 10:53:50 INFO: Analysis report uploaded in 76ms
> 10:53:50 INFO: ANALYSIS SUCCESSFUL, you can browse https://<REMOVED>/sonar/dashboard/index/Test.Customer.Service
> 10:53:50 INFO: Note that you will be able to access the updated dashboard once the server has processed the submitted analysis report
> 10:53:50 INFO: More about the report processing at https://<REMOVED>/sonar/api/ce/task?id=<REMOVED>
> 10:53:50 DEBUG: Report metadata written to /workspace/Test_Customer_Service/.sonar/report-task.txt
> 10:53:50 DEBUG: Post-jobs : org.sonar.plugins.buildbreaker.AlertBreaker@3c443976 -> org.sonar.plugins.buildbreaker.ForbiddenConfigurationBreaker@3e58d65e
> 10:53:50 INFO: Executing post-job org.sonar.plugins.buildbreaker.AlertBreaker
> 10:53:50 INFO: Executing post-job org.sonar.plugins.buildbreaker.ForbiddenConfigurationBreaker
> 10:53:50 INFO: Analysis report generated in /workspace/Test_Customer_Service/.sonar/batch-report
> 10:53:50 INFO: ------------------------------------------------------------------------
> 10:53:50 INFO: EXECUTION SUCCESS
> 10:53:50 INFO: ------------------------------------------------------------------------
> 10:53:50 INFO: Total time: 6.304s
> 10:53:50 INFO: Final Memory: 41M/103M
> 10:53:50 INFO: ------------------------------------------------------------------------
> 10:53:50 DEBUG: Execution getVersion
> 10:53:50 DEBUG: Execution stop
> 10:53:50 Archiving artifacts
> 10:54:01 Finished: SUCCESS
> 

**sonar.properties**

```
sonar.projectKey=Test.Customer.Service
sonar.projectName=TEST CUSTOMER SERVICE
sonar.projectVersion=1.0.0
sonar.projectDescription= Project test results.
 
sonar.ts.excludetypedefinitionfiles=true
sonar.ts.tslintconfigpath=tslint.json

sonar.sources=.
sonar.exclusions=node_modules/**,typings/**,test/record/**

sonar.language=ts
sonar.sourceEncoding=UTF-8

sonar.login=foo
sonar.password=foo
```

**Comment by Pablissimo** [comment]
Yeah that's weird, it looks like it's built a vaguely sensible command line but then struggles to use it. Will try to repro on a couple of linux hosts I've got knocking around, thanks for the report!

**Comment by gitawego** [comment]
I've exactly the same problem.

**Comment by gitawego** [comment]
I think if you want to handle folders with white-spaces, you can just add double quotes around the white-spaces. it works for both linux and windows.

**Comment by Pablissimo** [comment]
I wonder if just skipping absoluteifying the first argument would do the trick then - I know I have to supply quotes for subsequent paths otherwise things go horribly wrong on certain CI hosts (where the working directory is fairly arbitrary). Will try a couple of combinations - don't want to break TFS CI (which was basically all that 0.9 was trying to fix) but I'm sure I can figure it.

**Comment by gitawego** [comment]
I've tested with only removing the double quotes of first argument, it doesn't work. I've to remove all the double quotes in order to make it works. (while taking account the path with whitespaces)
if you wish, I can send a PR to let you review my modification.

**Comment by Pablissimo** [comment]
Yeah fire it on just to see - I just did the same but it then broke on Windows when the path has spaces

**Comment by Pablissimo** [comment]
Want to give this version a try? It's a daft temporary fix that quotes paths only a) on Windows and b) when there are spaces in the path. On Linux is should just skip all quoting, and appears to work on my local Debian installs but another tester would be good! Or compile from the issue47 branch.

[sonar-typescript-plugin-0.92-SNAPSHOT.zip](https://github.com/Pablissimo/SonarTsPlugin/files/440852/sonar-typescript-plugin-0.92-SNAPSHOT.zip)

**Comment by gitawego** [comment]
I just tested this build in unix (mac and ubuntu), it works. 
I didn't test yet for windows. (no windows at home...)

**Comment by chiefmc** [comment]
Same here. 
It was not working with `throw err`, but now woth v0.8 it doesn't work because of Sonar 6.0 :)

**Comment by Pablissimo** [comment]
Per the other SonarQube 6.0 issues, there's a pre-release version of an SQ6 supporting plugin that also includes this path fix - give it a bash and let me know!

https://github.com/Pablissimo/SonarTsPlugin/releases/tag/v0.93

(At your own risk of course, try a dev server first etc etc).

**Comment by Pablissimo** [comment]
Fixed in v0.94, please re-open if it still occurs.
